src/core/mod.rs

```diff
@@ -276,6 +291,70 @@ impl BPFBuilder {
             debug: self.debug,
         })
     }
+
+    #[cfg(any(
+        feature = "v0_18_0",
+        not(feature = "specific"),
+    ))]
+    /// Try constructing a BPF module from the builder
+    pub fn build(mut self) -> Result<BPF, BccError> {
+        let (code, contexts) = if self.usdt_contexts.is_empty() {
+            (self.code, Vec::new())
+        } else {
+            let (mut code, contexts) = usdt_generate_args(self.usdt_contexts)?;
+
+            let base_code = self.code.clone().to_str().map(|s| s.to_owned())?;
+            code.push_str(base_code.as_str());
+            let code = to_cstring(code.as_str())?;
+
+            (code, contexts)
+        };
+
+        let ptr = unsafe {
+            bpf_module_create_c_from_string(
+                code.as_ptr(),
+                self.debug.bits(),
+                self.cflags
+                    .iter()
+                    .map(|v| v.as_ptr())
+                    .collect::<Vec<*const c_char>>()
+                    .as_mut_ptr(),
+                self.cflags.len().try_into().unwrap(),
+                true,
+                self.device
+                    .as_ref()
+                    .map(|name| name.as_ptr())
+                    .unwrap_or(ptr::null_mut()),
+            )
+        };
+
+        if ptr.is_null() {
+            return Err(BccError::Compilation);
+        }
+
+        let mut bpf = BPF {
```

Comment:
I wonder if we want a helping function to construct this, most of the fields are really just `Default::default()`. Fair to consider out-of-scope for this PR.